jpf-symbc/src/main/gov/nasa/jpf/symbc/bytecode/SymbolicStringHandler.java

```diff
@@ -1185,19 +1220,48 @@ public void handleTrim(JVMInvokeInstruction invInst, ThreadInfo th) {
 		// throw new RuntimeException("ERROR: symbolic string method not Implemented - Trim");
 		StackFrame sf = th.getModifiableTopFrame();
 		StringExpression sym_v1 = (StringExpression) sf.getOperandAttr(0);
+
+		ChoiceGenerator<?> cg;
+		boolean conditionValue;
+		cg = th.getVM().getChoiceGenerator();
+
+		assert (cg instanceof PCChoiceGenerator) : "expected PCChoiceGenerator, got: " + cg;
+		conditionValue = (Integer) cg.getNextChoice() != 0;
 		int s1 = sf.pop();
+		PathCondition pc;
 
-		if (sym_v1 == null) {
-			ElementInfo e1 = th.getElementInfo(s1);
-			String val1 = e1.asString();
-			sym_v1 = new StringConstant(val1);
+		ChoiceGenerator<?> prev_cg = cg.getPreviousChoiceGenerator();
+		while (!((prev_cg == null) || (prev_cg instanceof PCChoiceGenerator))) {
+			prev_cg = prev_cg.getPreviousChoiceGenerator();
+		}
+
+		if (prev_cg == null)
+			pc = new PathCondition();
+		else
+			pc = ((PCChoiceGenerator) prev_cg).getCurrentPC();
+
+		assert pc != null;
+
+		if(conditionValue) {
+			if (sym_v1 == null) {
+				ElementInfo e1 = th.getElementInfo(s1);
+				String val1 = e1.asString();
+				sym_v1 = new StringConstant(val1);
+			}
+		} else {
+			pc.spc._addDet(StringComparator.EQUALS, sym_v1, "null");
+			if (!pc.simplify()) {
+				th.getVM().getSystemState().setIgnored(true);
+			} else {
+				th.createAndThrowException("java.lang.NullPointerException");
```

Comment:
all these check should be checked if your `npe` flag is true, we only add then the null checks to the pc and check them, otherwise we ignore. this should happen to all functions where we raise those expections similar to what we have done for the `contains`. It is not something that is special to contains only, all string functions that could raise a null pointer exception should include this check.